test/ui/shared/test-url-util.js

```diff
@@ -0,0 +1,61 @@
+/* global describe, it */
+
+import expect from 'expect';
+import UrlUtil from '../../../ui/shared/url-util';
+
+describe('UrlUtil', function () {
+  describe('getHostname', function() {
+    it('returns hostname for full url', function() {
+      expect(UrlUtil.getHostname('http://example.com')).toEqual('example.com');
+    });
+    it('returns null for empty url', function() {
+      expect(UrlUtil.getHostname('')).toEqual(null);
+    });
+  });
+
+  describe('getScheme', function() {
+    it('is null for empty', function() {
+      expect(UrlUtil.getScheme('/file/path/to/file')).toEqual(null);
+    });
+
+    it('is null for localhost', function() {
+      expect(UrlUtil.getScheme('localhost://127.0.0.1')).toEqual(null);
```

Comment:
There's special case in the imported url utils for `scheme === 'localhost://'`.  I'm not sure why